indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexIngestionSpec.java

```diff
@@ -45,9 +44,7 @@ public ParallelIndexIngestionSpec(
     super(dataSchema, ioConfig, tuningConfig);
 
     if (dataSchema.getParserMap() != null && ioConfig.getInputSource() != null) {
-      if (!(ioConfig.getInputSource() instanceof FirehoseFactoryToInputSourceAdaptor)) {
-        throw new IAE("Cannot use parser and inputSource together. Try using inputFormat instead of parser.");
-      }
+      throw new IAE("Cannot use parser and inputSource together. Try using inputFormat instead of parser.");
```

Comment:
what if the parserMap is not null and input source is an adapter input source?

Reply:
input source here can't be an instance of adapter input source since the user can't create it directly, and it's generated by `IndexTask.IndexIOConfig.getNonNullInputSource()` from `InputRowParser` & `FiniteFirehoseFactory` previously in case `InputSource` is null, now that `FiniteFirehoseFactory` is removed user has to submit non-null input source for batch index tasks for all the cases where FiniteFirehoseFactory instantiating spec was used.